src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -264,6 +276,42 @@ class EthService(
     }
   }
 
+  def getCode(req: GetCodeRequest): ServiceResponse[GetCodeResponse] = {
+    Future.successful {
+      resolveBlock(req.block).map { block =>
+        val world = InMemoryWorldStateProxy(blockchainStorages, Some(block.header.stateRoot))
```

Comment:
I'm wondering, although it is quite easy to use an `InMemoryWorldStateProxy`, as it has a structure and design around it for saving changes before persisting them (for the `Ledger`), should we only use a simple MPT for this? Maybe we could have also a `WorldState` without this extra cache? This shouldn't be a blocker for merging this PR though